docs/raw/serving-community-history.md

```diff
@@ -0,0 +1,315 @@
+
+Community owner nodes SHOULD remove those messages from their local databases after they have been turned into archives and distributed to the BitTorrent network.
+
+## Exporting messages for bundling
+
+Community owner nodes export messages from their local database for creating and bundling history archives using the following criteria:
+
+- Messages to be exported MUST have a `contentTopic` that matches any of the topics of the community channels
+- Messages to be exported MUST have a `timestamp` that lies within a given time range
+
+The range for the `timestamp` depends on the context in which the community owner node attempts to create a history archive. This can be one of the following:
+
+1. The community owner node attempts to create an archive periodically for the past seven days (including the current day). In this case, the `timestamp` has to lie within the day the last time an archive was created and the current day.
+2. The community owner node has been offline and attempts to create an archive for all the live messages it has missed since it went offline. In this case, the `timestamp` has to lie within the day the latest message was received and the current day.
+
+Exported messages MUST be restored as [14/WAKU2-MESSAGE](https://rfc.vac.dev/spec/14/) for bundling. Waku messages that have been exported for bundling can now be removed from the community owner node's database (community owner nodes still maintain a database of application messages).
+
+## Message history archives
+
+Message history archives are represented as `WakuMessageArchive` and created from messages exported from the local database. Message history archives are implemented using the following protocol buffer.
+
+### WakuMessageHistoryArchive
+
+The `from` field SHOULD contain a timestamp of the time range's lower bound.
+
+The `to` field SHOULD contain a timestamp of the time range's the higher bound.
+
+The `contentTopic` field MUST contain the same `contentTopic` that the archive's `messages` have.
+
+The `messages` field MUST contain all messages that belong into the archive given its `from`, `to` and `contentTopic` fields.
+
+```
+syntax = "proto3"
+
+message WakuMessageArchive {
+  uint64 from = 1
+  uint64 to = 2
+  string contentTopic = 3
+  repeated WakuMessage messages = 4 // `WakuMessage` is provided by 14/WAKU2-MESSAGE
+}
+```
+
+## Creating magnet links
+
+Once a message archive is created, the community owner node MUST derive a magnet link following the [Magnet URI scheme](https://en.wikipedia.org/wiki/Magnet_URI_scheme) using the underlying BitTorrent protocol client.
+
+The `dn` parameter ("display name") in the resulting magnet link MAY be optional.
+
+The resulting magnet link MUST be bundled into a `WakuMessageArchiveIndex`, which is then later distributed to other Status nodes.
```

Comment:
@PascalPrecht I don't understand this step, why must the magnet link be bundled into WakuMessageArchiveIndex?  I would have thought it just needs to be sent to all other community members via the hidden channel, but perhaps I'm missing something?

Reply:
I've tried to explain this here: https://github.com/status-im/specs/pull/162/files#r767957001

Then "bundling" in this spec merely refers to:

"Let's create an index of magnet links including all magnet links that have been created in the past + the new one that was just created"

^ That thing (also known as `WakuMessageArchiveIndex`) is then sent to community members via the special channel.

We can decide if we want to send this as-is, or, if we should create a magnet link for that index as well. So, sending the index directly vs sending the index as magnet link.

Mentioned it here: https://github.com/status-im/specs/pull/162#discussion_r767717175